### PR TITLE
Bugfix/DatePicker id for input#1812

### PR DESCRIPTION
**Pull request by vik753**
### Description: [DatePicker]: passed id prop to the TextInput, [PickerToggler]: passed id prop to the input through PickerInput. [RangeDatePicker]: set id prop for first input, [DemoForm]: added id for inputs with label. [LabeledInput Demo]: set htmlFor and id for labels & inputs. [FilterNumericBody]: fixed isWrongRange function.

### CODE REVIEW

File: uui/components/datePickers/DatePicker.tsx [hunk 1]
```diff
@@ -10,7 +10,12 @@ import { DropdownContainer } from '../overlays';
 const defaultMode = EditMode.FORM;
 
 /** Represents the properties of the DatePicker component. */
-export interface DatePickerProps extends DatePickerCoreProps, SizeMod, IHasEditMode {}
+export interface DatePickerProps extends DatePickerCoreProps, SizeMod, IHasEditMode {
+    /**
+    * Defines id for inner input
```
Comment (AlekseyManetov) [inline uui/components/datePickers/DatePicker.tsx:15]: Let's change to - 
    /** HTML ID attribute for the toggler input */

File: uui/components/datePickers/RangeDatePicker.tsx [hunk 2]
```diff
@@ -19,6 +19,10 @@ export interface RangeDatePickerProps extends BaseRangeDatePickerProps, SizeMod
      * A pure function that gets placeholder for 'from' or 'to' input.
      */
     getPlaceholder?(type: InputType): string;
+    /**
+    * Defines id for inner input
+    */
+    id?: string;
```
Comment (AlekseyManetov) [inline uui/components/datePickers/RangeDatePicker.tsx:25]: Let's change to -
/** HTML ID attribute for the first input into toggler */

File: uui/components/pickers/PickerInput.tsx [hunk 3]
```diff
@@ -14,7 +14,12 @@ import { PickerItem } from './PickerItem';
 const pickerHeight = 300;
 const pickerWidth = 360;
 
-export type PickerInputProps = SizeMod & IHasEditMode & {};
+export type PickerInputProps = SizeMod & IHasEditMode & {
+    /**
+    * Defines an id for inner input
+    */
+    id?: string;
```
Comment (AlekseyManetov) [inline uui/components/pickers/PickerInput.tsx:21]: Let's change to -
/** HTML ID attribute for the toggler input */

File: uui/components/pickers/PickerInput.tsx [hunk 4]
```diff
@@ -178,7 +183,7 @@ function PickerInputComponent<TItem, TId>({ highlightSearchMatches = true, ...pr
         <Dropdown
             renderTarget={ (dropdownProps) => {
                 const targetProps = getTogglerProps();
-                return renderTarget({ ...dropdownProps, ...targetProps });
+                return renderTarget({ id: props?.id, ...dropdownProps, ...targetProps });
```
Comment (AlekseyManetov) [inline uui/components/pickers/PickerInput.tsx:186]: Can we move 'id' prop defining to the targetProps?

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui/components/datePickers/DatePicker.tsx (hunk 1 above)
Comment (vik753) [reply uui/components/datePickers/DatePicker.tsx:15]: fixed

File: uui/components/datePickers/RangeDatePicker.tsx (hunk 2 above)
Comment (vik753) [reply uui/components/datePickers/RangeDatePicker.tsx:25]: fixed

File: uui/components/pickers/PickerInput.tsx (hunk 3 above)
Comment (vik753) [reply uui/components/pickers/PickerInput.tsx:21]: fixed

File: uui/components/pickers/PickerInput.tsx (hunk 4 above)
Comment (vik753) [reply uui/components/pickers/PickerInput.tsx:186]: fixed

File: uui/components/pickers/PickerToggler.tsx [hunk 5]
```diff
@@ -68,6 +72,7 @@ function PickerTogglerComponent<TItem extends string, TId>(props: PickerTogglerP
     return (
         <UuiPickerToggler
             { ...props }
+            id={ props?.id }
```
Comment (AlekseyManetov) [inline uui/components/pickers/PickerToggler.tsx:75]: Is it this props will no be spreaded by { ...props }?

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui/components/pickers/PickerToggler.tsx (hunk 5 above)
Comment (vik753) [reply uui/components/pickers/PickerToggler.tsx:75]: fixed

Comment (AlekseyManetov) [reply uui/components/pickers/PickerToggler.tsx:75]: still doesn't fixed

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui/components/pickers/PickerToggler.tsx (hunk 5 above)
Comment (vik753) [reply uui/components/pickers/PickerToggler.tsx]: sorry, already fixed

File: uui-components/src/pickers/hooks/types.ts [hunk 6]
```diff
@@ -92,6 +92,9 @@ IHasIcon & {
 
     /** Search input debounce delay in ms. Default value is 500ms */
     searchDebounceDelay?: number;
+
+    /** HTML ID attribute for the picker input */
```
[the remark this answers is not in the document]
Comment (vik753) [reply uui-components/src/pickers/hooks/types.ts:96]: fixed

**Pull request merged**